iceoryx_hoofs/include/iceoryx_hoofs/memory/relative_pointer.inl

```diff
@@ -239,11 +230,71 @@ inline typename RelativePointer<T>::ptr_t RelativePointer<T>::computeRawPtr() co
     return getPtr(segment_id_t{m_id}, m_offset);
 }
 
+// AXIVION Next Construct AutosarC++19_03-A15.5.3, AutosarC++19_03-A15.4.2, FaultDetection-NoexceptViolations : False
+// positive, std::terminate is not called in the c'tor of PointerRepository and noexcept-specification is not violated
 inline PointerRepository<segment_id_underlying_t, UntypedRelativePointer::ptr_t>& getRepository() noexcept
 {
+    // AXIVION Next Construct AutosarC++19_03-A3.3.2 : PointerRepository can't be constexpr, usage of the static
+    // object is encapsulated in the RelativePointer class
     static PointerRepository<segment_id_underlying_t, UntypedRelativePointer::ptr_t> repository;
     return repository;
 }
+
+template <typename T>
+// AXIVION Next Line AutosarC++19_03-A13.5.5 : The RelativePointer shall explicitly be comparable to raw pointers
+inline bool operator==(const RelativePointer<T> lhs, T* const rhs) noexcept
+{
+    return lhs.get() == rhs;
+}
+
+template <typename T>
+// AXIVION Next Line AutosarC++19_03-A13.5.5 : The RelativePointer shall explicitly be comparable to raw pointers
+inline bool operator==(T* const lhs, const RelativePointer<T> rhs) noexcept
+{
+    return rhs == lhs;
+}
+
+template <typename T>
+// AXIVION Next Line AutosarC++19_03-A13.5.5 : The RelativePointer shall explicitly be comparable to nullptrs
+inline bool operator==(std::nullptr_t, const RelativePointer<T> rhs) noexcept
+{
+    return rhs.get() == nullptr;
+}
+
+template <typename T>
+// AXIVION Next Line AutosarC++19_03-A13.5.5 : The RelativePointer shall explicitly be comparable to nullptrs
+inline bool operator==(const RelativePointer<T> lhs, std::nullptr_t) noexcept
+{
+    return lhs.get() == nullptr;
+}
+
+template <typename T>
+// AXIVION Next Line AutosarC++19_03-A13.5.5 : The RelativePointer shall explicitly be comparable to raw pointers
+inline bool operator!=(const RelativePointer<T> lhs, T* const rhs) noexcept
+{
+    return !(lhs == rhs);
+}
+
+template <typename T>
+// AXIVION Next Line AutosarC++19_03-A13.5.5 : The RelativePointer shall explicitly be comparable to raw pointers
+inline bool operator!=(T* const lhs, const RelativePointer<T> rhs) noexcept
+{
+    return rhs != lhs;
+}
+
+template <typename T>
+// AXIVION Next Line AutosarC++19_03-A13.5.5 : The RelativePointer shall explicitly be comparable to nullptrs
+inline bool operator!=(std::nullptr_t, const RelativePointer<T> rhs) noexcept
+{
+    return rhs.get() != nullptr;
```

Comment:
Suggestion:
Could in theory be done in terms of the equality operator (and would be inlined), but I also prefer it as is.